lib/msal-browser/src/broker/EmbeddedClientApplication.ts

```diff
@@ -4,72 +4,107 @@
+import { Logger, ClientAuthError, AuthenticationResult, CacheRecord, CacheManager, AccessTokenEntity, IdTokenEntity, AccountEntity } from "@azure/msal-common";
 import { BrokerHandshakeRequest } from "./BrokerHandshakeRequest";
 import { BrokerHandshakeResponse } from "./BrokerHandshakeResponse";
 import { PopupRequest } from "../request/PopupRequest";
 import { RedirectRequest } from "../request/RedirectRequest";
 import { BrokerAuthRequest } from "./BrokerAuthRequest";
 import { InteractionType } from "../utils/BrowserConstants";
 import { BrokerRedirectResponse } from "./BrokerRedirectResponse";
+import { BrokerAuthResult } from "./BrokerAuthResult";
+import { BrowserStorage } from "../cache/BrowserStorage";
 import { BrowserAuthError } from "../error/BrowserAuthError";
 
 const DEFAULT_MESSAGE_TIMEOUT = 2000;
 /**
  * Embedded application in a broker scenario.
  */
-export class BrokerClient {
+export class EmbeddedClientApplication {
     private brokerOptions: BrokerOptions;
     private logger: Logger;
     private clientId: string;
     private version: string;
     private brokerOrigin: string;
+    private browserStorage: BrowserStorage;
 
+    private get trustedBrokersProvided(): boolean {
+        return this.brokerOptions.trustedBrokerDomains && this.brokerOptions.trustedBrokerDomains.length >= 1;
+    };
     public brokeringEnabled: boolean;
 
-    constructor(brokerOptions: BrokerOptions, logger: Logger, clientId: string, version: string) {
+    constructor(brokerOptions: BrokerOptions, logger: Logger, clientId: string, version: string, browserStorage: BrowserStorage) {
         this.brokerOptions = brokerOptions;
         this.logger = logger;
         this.clientId = clientId;
         this.version = version;
-
-        if (!this.brokerOptions.trustedBrokerDomains || this.brokerOptions.trustedBrokerDomains.length < 1) {
-            this.logger.info("Application was identified as an embedded app in a broker scenario, but no trusted broker domains were provided.");
-            this.brokeringEnabled = false;
-        } else {
-            this.brokeringEnabled = true;
-        }
+        this.browserStorage = browserStorage;
+        this.brokeringEnabled = false;
     }
 
     async initiateHandshake(): Promise<void> {
-        if (!this.brokeringEnabled) {
-            this.logger.verbose("Brokering is not enabled, handshake was not performed.");
-            return;
+        if (!this.trustedBrokersProvided) {
+            throw BrowserAuthError.createNoTrustedBrokersProvidedError();
         }
 
         try {
             const response = await this.sendHandshakeRequest();
             this.brokerOrigin = response.brokerOrigin;
+            this.brokeringEnabled = true;
         } catch (e) {
             this.logger.error(e);
             this.brokeringEnabled = false;
             throw e;
         }
     }
 
+    private async preflightBrokerRequest(): Promise<void> {
+        if (!this.brokeringEnabled) {
```

Comment:
Can we call this something like `brokerCommunicationEstablished` ? I was initially confused why we were checking for falsey `brokeringEnabled`

Reply:
done.